pkg/ccl/sqlccl/backup.go

```diff
@@ -553,6 +557,65 @@ func backupPlanHook(
 	return fn, header, nil
 }
 
+func showBackupPlanHook(
+	baseCtx context.Context, stmt parser.Statement, p sql.PlanHookState,
+) (func() ([]parser.Datums, error), sqlbase.ResultColumns, error) {
+	backup, ok := stmt.(*parser.ShowBackup)
+	if !ok {
+		return nil, nil, nil
+	}
+
+	if err := utilccl.CheckEnterpriseEnabled("SHOW BACKUP"); err != nil {
+		return nil, nil, err
+	}
+
+	if err := p.RequireSuperUser("SHOW BACKUP"); err != nil {
+		return nil, nil, err
+	}
+
+	toFn, err := p.TypeAsString(&backup.Path)
+	if err != nil {
+		return nil, nil, err
+	}
+	header := sqlbase.ResultColumns{
+		{Name: "database", Typ: parser.TypeString},
+		{Name: "table", Typ: parser.TypeString},
+		{Name: "start_time", Typ: parser.TypeTimestamp},
+		{Name: "end_time", Typ: parser.TypeTimestamp},
+	}
+	fn := func() ([]parser.Datums, error) {
+		str := toFn()
+		desc, err := readBackupDescriptor(baseCtx, str)
```

Comment:
Not: I think we want to make our own span here, similar to above, and the pass that rather than baseCtx.